From 2543dd4f9f47f94c1b7850ef2f24bb5fa2aac5b1 Mon Sep 17 00:00:00 2001
From: Karl Cheng <karl@canva.com>
Date: Tue, 26 Apr 2022 18:15:51 +1000
Subject: [PATCH] Test

---
 internal/npm_install/npm_install.bzl | 27 +++++++++++++++++++++++++++
 1 file changed, 27 insertions(+)

diff --git internal/npm_install/npm_install.bzl internal/npm_install/npm_install.bzl
index a56bdbc8..69b76bbf 100644
--- internal/npm_install/npm_install.bzl
+++ internal/npm_install/npm_install.bzl
@@ -788,6 +788,33 @@ cd /D "{root}" && "{yarn}" {yarn_args}
     _symlink_node_modules(repository_ctx)
     _apply_post_install_patches(repository_ctx)
 
+    result = repository_ctx.execute([
+        "find",
+        repository_ctx.path("node_modules"),
+        "-type",
+        "f",
+        "-name",
+        "*.pyc",
+        "-delete",
+    ])
+    if result.return_code:
+        fail("deleting .pyc files failed: %s (%s)" % (result.stdout, result.stderr))
+    result = repository_ctx.execute([
+        "find",
+        repository_ctx.path("node_modules"),
+        "-type",
+        "f",
+        "-name",
+        "*.node",
+        "-exec",
+        "strip",
+        "-S",
+        "{}",
+        ";",
+    ])
+    if result.return_code:
+        fail("stripping .node files failed: %s (%s)" % (result.stdout, result.stderr))
+
     _create_build_files(repository_ctx, "yarn_install", node, repository_ctx.attr.yarn_lock, repository_ctx.attr.generate_local_modules_build_files)
 
 yarn_install = repository_rule(
-- 
2.35.1

